From b1f26dc76158e1f3364836d8b540b17a7a1ccb3d Mon Sep 17 00:00:00 2001
From: Tom de Vries <tdevries@suse.de>
Date: Sat, 20 Mar 2021 09:37:31 +0100
Subject: [PATCH] [testsuite] Handle readelf following links by default

Since binutils commit c46b706620e "Change the readelf and objdump programs so
that they will automatically follow links to separate debug info files" there
are a few FAILs in the dwz testsuite.

This is just due to changing the default behaviour for readelf.

Fix this by testing whether the new -Wn readelf flag is supported,
and if so, adding it to the readelf call.

2021-03-20  Tom de Vries  <tdevries@suse.de>

	PR dwz/27592
	* testsuite/dwz.tests/odr-struct-multifile.sh: Call readelf with -wN
	if supported.
	* testsuite/dwz.tests/cycle.sh: Same.
---
 testsuite/dwz.tests/odr-struct-multifile.sh | 9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/testsuite/dwz.tests/odr-struct-multifile.sh b/testsuite/dwz.tests/odr-struct-multifile.sh
index cc462c9..20b682b 100644
--- a/testsuite/dwz.tests/odr-struct-multifile.sh
+++ b/testsuite/dwz.tests/odr-struct-multifile.sh
@@ -2,6 +2,11 @@ if ! $execs/dwz-for-test --odr -v 2>/dev/null; then
     exit 77
 fi
 
+readelf_flags=""
+if readelf -h 2>&1 | grep -q "\-wN"; then
+    readelf_flags=-wN
+fi
+
 cp $execs/odr-struct 1
 cp 1 2
 
@@ -41,12 +46,12 @@ done
 
 
 for name in aaa bbb ccc; do
-    cnt=$(readelf -wi 1 | grep -c "DW_AT_name.*:.*$name" || true)
+    cnt=$(readelf -wi $readelf_flags 1 | grep -c "DW_AT_name.*:.*$name" || true)
     [ $cnt -eq 0 ]
 done
 
 for name in member_one member_two member_three member_four; do
-    cnt=$(readelf -wi 1 | grep -c "DW_AT_name.*:.*$name" || true)
+    cnt=$(readelf -wi $readelf_flags 1 | grep -c "DW_AT_name.*:.*$name" || true)
     [ $cnt -eq 0 ]
 done
 
-- 
1.8.3.1

